contrib/natss/pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -144,9 +150,10 @@ func (s *SubscriptionsSupervisor) subscribe(channel provisioners.ChannelReferenc
 
 	mcb := func(msg *stan.Msg) {
 		s.logger.Sugar().Infof("NATSS message received from subject: %v; sequence: %v; timestamp: %v", msg.Subject, msg.Sequence, msg.Timestamp)
-		message := provisioners.Message{
-			Headers: map[string]string{},
-			Payload: []byte(msg.Data),
+		message := provisioners.Message{}
+		if err := json.Unmarshal(msg.Data, &message); err != nil {
```

Comment:
Unit test.

Reply:
Since I just do here basic Marshal/Unmarshal without much of processing, appreciate some ideas what do unit test. thank you.

Reply:
A test which manually writes malformed data to the subscription (e.g. a string with a missing `}`) should be able to validate that the message was propagated but discarded due to corruption.